server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -173,6 +180,7 @@ protected int primaryOperationCount(BulkShardRequest request) {
         return request.items().length;
     }
 
+    // TODO PG this is just for testing?
```

Comment:
TODO?

Reply:
I was worried that there is another public method `performOnPrimary` but it turns out that it is only used in testing. It is being called by xpack test, hence a scope had to be changed.